src/VecSim/algorithms/brute_force.cpp

```diff
@@ -118,6 +118,22 @@ static void BruteForce_UpdateVector(BruteForceIndex *bfIndex, idType id, const v
     memcpy(destinaion, vector_data, bfIndex->base.dim);
 }
 
+static size_t BruteForce_IndexOfMin(float *scores, size_t n) {
+    float max_float = std::numeric_limits<float>::max();
+    float inf = numeric_limits<float>::infinity();
+    float min = std::numeric_limits<float>::max();
+    size_t min_index = 0;
+    for (size_t i = 0; i < n; i++) {
+        if (scores[i] == max_float)
+            continue;
+        if (scores[i] < min || scores[i] == inf) {
```

Comment:
Why we set `scores[i]` as the minimum score if `scores[i] == inf`?

Reply:
this is awkward logic:
since there is an `OR` evaluation here, if the first condition is `true`, so the entire condition is `true`. The first test is for actual numbers. Actual numbers can be either the minimal number in the array, or not.
For a number smaller from the current `min`, the LHS of the condition is `true`, otherwise, the entire condition is `false`.
After a number is found is the current minimum fo the array, it is replaced with `std::numeric_limits<float>::max()`. Once we replaced all normal numbers we are left with `inf` or `std::numeric_limits<float>::max()`, so in case we have `inf`, RHS of this condition will hold, and will be replaced with `std::numeric_limits<float>::max()`